src/vswhere.lib/LegacyProvider.cpp

```diff
@@ -9,12 +9,28 @@ using namespace std;
 
 ILegacyProvider& LegacyProvider::Instance = LegacyProvider();
 
+LegacyProvider::LegacyProvider() :
+    m_hKey32(NULL), m_hKey64(NULL)
+{
+    auto err = ::RegOpenKeyExW(HKEY_LOCAL_MACHINE, L"SOFTWARE\\Microsoft\\VisualStudio\\SxS\\VC7", 0, KEY_QUERY_VALUE | KEY_WOW64_32KEY, &m_hKey32);
```

Comment:
This really is outside the scope of what was intended for vswhere. It also does not handle all the legacy versions of Visual Studio that vswhere partly supports.